src/main/java/org/opensearch/security/support/Base64JDKHelper.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015-2018 _floragunn_ GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.support;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidClassException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.ObjectStreamClass;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.BaseEncoding;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.core.common.Strings;
+
+import static org.opensearch.security.support.SafeSerializationUtils.isSafeClass;
+
+public class Base64JDKHelper {
+
+    private final static class SafeObjectOutputStream extends ObjectOutputStream {
+
+        private static final boolean useSafeObjectOutputStream = checkSubstitutionPermission();
+
+        @SuppressWarnings("removal")
+        private static boolean checkSubstitutionPermission() {
+            SecurityManager sm = System.getSecurityManager();
+            if (sm != null) {
+                try {
+                    sm.checkPermission(new SpecialPermission());
+
+                    AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+                        AccessController.checkPermission(SUBSTITUTION_PERMISSION);
```

Comment:
Do we need this? For context, this was used when classes were replaced in earlier versions.

Reply:
I've not touched the existing implementation as part of this PR.